src/compiler/types.ts

```diff
@@ -726,6 +727,7 @@ namespace ts {
     // @kind(SyntaxKind.StringKeyword)
     // @kind(SyntaxKind.SymbolKeyword)
     // @kind(SyntaxKind.VoidKeyword)
+    // @kind(SyntaxKind.ThisType)

```

Comment:
I think there would need to be a `StringLiteralType` kind and a `StringLiteral` kind, just like how there's `ThisType` and `ThisKeyword`. So maybe a followup PR?